site/app/libraries/database/DatabaseQueries.php

```diff
@@ -764,14 +779,14 @@ public function getUnviewedPosts($thread_id, $user_id) {
         return $rows;
     }
 
-    public function createThread($markdown, $user, $title, $content, $anon, $prof_pinned, $status, $hasAttachment, $categories_ids, $lock_thread_date, $expiration) {
-
+    public function createThread($markdown, $user, $title, $content, $anon, $prof_pinned, $status, $hasAttachment, $categories_ids, $lock_thread_date, $expiration, $announcement) {
         $this->course_db->beginTransaction();
 
+        $now = $announcement ? new \DateTime("now") : null;
```

Comment:
It might be better to use $this->core->getDateTimeNow() since it uses the config's timezone.